### ISSUE TITLE
Cannot find runtime 'node' on PATH

### ISSUE THREAD

**Issue by tapir** [Microsoft/vscode#10571]
- VSCode Version: 1.4.0
- OS Version: Windows 7 64bit

Steps to Reproduce:
1. Try to run a node app

As discussed here https://github.com/Microsoft/vscode-react-native/issues/284
I've triple/quadruple checked that `node.exe` is on my path. I can run it via command line just by typing "node".
I've also cleaned `.vscode` directory in `C:\Users\MyUser` just to reset everything. That did not work as well.

**Comment by tapir** [comment]
* Wiped everything in my computer that is related to VSCode and nodejs/npm that I could find. * Installed VSCode insider and nodejs 6.3.1
* Put C:\Program Files\nodejs to my path.
* Restarted computer. 
* Opened up cmd.exe and tested if node works.
* Opened up VSCode Insider. Created a simple node app. Tried to launch it.

Still getting the same error.

**Comment by weinand** [comment]
@tapir does it make a difference if you launch VS Code from a command prompt?

**Comment by tapir** [comment]
@weinand It didn't work

**Comment by tapir** [comment]
Here is a screencast showing it: <iframe width="853" height="480" src="//sendvid.com/embed/b40amclw" frameborder="0" allowfullscreen></iframe>

**Comment by weinand** [comment]
Could you please compare the value of the PATH system variable inside VS Code with the value you see outside. For this open the 'Integrated Terminal' in VS Code and run `echo %PATH%`.

**Comment by tapir** [comment]
There you go
![external](https://cloud.githubusercontent.com/assets/223984/17696060/ab46b008-63b4-11e6-8e97-a7c33b5041d5.PNG)
![internal](https://cloud.githubusercontent.com/assets/223984/17696061/ab87c3d6-63b4-11e6-8e97-14b623e3bb0b.PNG)

**Comment by weinand** [comment]
@tapir Since the paths seem to be identical, I have no idea why the debugger cannot find node.
I would do a reboot in this situation.
If that doesn't help you can always specify the absolute path to node in the `runtimeExecutable` attribute.

**Comment by tapir** [comment]
@weinand That works but only for nodejs. How would I set the runtimeExecutable for react-native-tools? It doesn't accept that property.

**Comment by tapir** [comment]
Here is another problem which I think is related: VSCode can't run "taskkill" command which is a windows cmd command and can be executed outside of VSCode, in a cmd session.

**Comment by weinand** [comment]
If this problem would be a fundamental problem with VS Code I would expect more than one error report about this. But I'm not aware of any similar error. So something seems to be strange with your system.

Can you try to run VS Code without extensions? From the command line, execute: `code --disable-extensions` and try your steps again to see if it reproduces.

If that does not make a difference please try this:
- set your runtimeExecutable attribute to the node path
- start debugging and hit a breakpoint or stop on entry
- in the debug console type `process.env.PATH` and compare it again with the PATH from outside.

**Comment by tapir** [comment]
Did exactly that. First one did not work. Second one has the correct PATH on debug mode.
I agree that this seem to affect only me.

I believe, though, it's related to the debug part of the VSCode. My reasoning is that:
* I can run node from integrated terminal.
* I can run a react app via React: Run Android command provided by the plugin.

BUT, when I try to do the same from the debug window, it can't find node.

**Comment by weinand** [comment]
@tapir what do you mean by "First one did not work"? You cannot reproduce the problem if you disable all extensions?

**Comment by tapir** [comment]
I mean I still get the same error.

**Comment by weinand** [comment]
@tapir what is the result when you run `where node` in a command prompt?

**Comment by tapir** [comment]
Wow. Indeed something was wrong with my system because cmd couldn't find the command "where". That got me to see that the global (not per user one) PATH var was completely deleted.

I've searched for the default windows 7 PATH value which was this: `%SystemRoot%\system32;%SystemRoot%;%SystemRoot%\System32\Wbem;%SYSTEMROOT%\System32\WindowsPowerShell\v1.0\`

When I've restored this, everything worked perfectly.
I must ask though, what does VSCode uses that depends on this?
